### PR TITLE
SOC-525 | Cleanup in CaptchaController

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-525

### CODE REVIEW

File: extensions/wikia/Captcha/CaptchaController.class.php [hunk 1]
```diff
@@ -5,18 +5,12 @@
...
 	 * Displays a captcha image. This is used exclusively by FancyCaptcha.
 	 */
 	public function showImage() {
 		$fancyCaptcha = new \Captcha\Module\FancyCaptcha();
-		if ( !$fancyCaptcha->showImage() ) {
-			$this->response->setData( [
-				'error' => wfMessage( 'captcha-no-image' )->escaped(),
-			] );
-		}
+		$fancyCaptcha->showImage();
```
Comment (kvas-damian) [inline extensions/wikia/Captcha/CaptchaController.class.php]: We don't need to put additional error message. It's already handled: https://github.com/Wikia/app/blob/dev/extensions/wikia/Captcha/Module/FancyCaptcha.class.php#L260

File: extensions/wikia/Captcha/CaptchaController.class.php [hunk 2]
```diff
@@ -29,11 +23,4 @@ public function getFancyCaptcha() {
 			[ 'form' => $fancyCaptcha->getForm() ]
 		);
 	}
-
-	/**
-	 * Display information about how this captcha works
-	 */
-	public function showHelp() {
```
Comment (kvas-damian) [inline extensions/wikia/Captcha/CaptchaController.class.php]: I don't think we use this method. When I tried to call it, I got empty page

**Comment by bkoval** [comment]
LGTM

File: extensions/wikia/Captcha/CaptchaController.class.php [hunk 3]
```diff
@@ -5,17 +5,17 @@
...
-		if ( !$fancyCaptcha->showImage() ) {
-			$this->response->setData( [
-				'error' => wfMessage( 'captcha-no-image' )->escaped(),
-			] );
+		if ( $fancyCaptcha->showImage() ) {
+			/*
+			* Stop immediately when captcha is sent to browser.
+			* We don't want to send additional headers and other nirvana stuff
+			*/
+			exit();
```
Comment (rafalkalinski) [inline extensions/wikia/Captcha/CaptchaController.class.php]: I don't think there are good grounds for exit like that.
I know that this is quick solution for PHP Warning we currently have, but we should at least create a new ticket for fixing/changing the controller to be able not sending headers when it should not.

Comment (kvas-damian) [inline extensions/wikia/Captcha/CaptchaController.class.php]: https://wikia-inc.atlassian.net/browse/SOC-583

**Pull request merged**